mozci/push.py

```diff
@@ -236,6 +239,10 @@ def tasks(self):
         Returns:
             list: A list of `Task` objects.
         """
+        push_tasks = adr.config.cache.get(self.push_uuid)
+        if push_tasks:
+            logger.info("Fetched tasks for {} from the cache".format(self.push_uuid))
+            return push_tasks
```

Comment:
I've got a bit carried away and I'm caching all Task instances.
I am only concerned I've cached more than I had been asked to. I'm happy to go back to just storing the groups, errors and results.

Reply:
There are two things we should take into account:
- is caching the whole task going to result in a too large cache compared to just caching groups, errors and results?
- ADR already caches part of the same data (or the same data, when groups are available in unittest), so we should disable caching for the adr queries in this function, or we should only do our caching when the data is not available in AD and we had to load it from Taskcluster; if we don't do this, we'd cache twice the data for many pushes.